server/src/main/resources/rules/rules.js

```diff
@@ -3082,14 +3073,14 @@ var Autobind = {
                 // Only really consider the group if it provides a necessary product, role, addon,
                 // or stacks with an existing partial stack
                 if (this.get_common_products(installed, ent_group).length > 0 ||
-                    this.get_common_role(role, ent_group).length > 0 ||
-                    this.get_common_addons(addons, ent_group).length > 0 ||
+                    ent_group.get_common_role(role) != null ||
+                    ent_group.get_common_addons(addons).length > 0 ||
                     (ent_group.attached_ents !== null && ent_group.attached_ents.length > 0)) {
                     valid_groups.push(ent_group);
                     ent_group.remove_extra_attrs(role, addons);
                     ent_group.prune_pools(role, addons);
                 } else {
-                    log.debug("Group " + ent_group.stack_id + " provides no needed products");
+                    log.debug("Group " + ent_group.stack_id + " provides no needed products, roles or addons");
```

Comment:
Try to get into the habit of using the format syntax rather than concatenation for these.